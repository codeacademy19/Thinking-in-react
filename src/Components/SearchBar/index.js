import React,{Component} from 'react';
class SearchBar extends Component {
   
    render() {
      return (
        <form>
          <input type="text" className="TextArea" 
            placeholder="Search..."
            value={this.props.filterText}
            onChange={(e)=>{this.props.handleFilterTextChange(e.target.value)}}
          />
          <p>
            <input className="CheckBox"
              type="checkbox"
              checked={this.props.inStockOnly}
              onChange={(e)=>{this.props.handleInStockChange(e.target.checked)}}
            />
            {' '}
            Only show products in stock
          </p>
        </form>
      );
    }
    
  }
  
  export default SearchBar;