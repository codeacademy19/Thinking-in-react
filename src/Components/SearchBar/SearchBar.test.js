import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '.';
import {shallow} from 'enzyme';


describe('SearchBar',()=>{
    it('should render a Search box and a check box',()=>{
        const tree=renderer.create(<SearchBar/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('should trigger handleFilterTextChange on changing the textbox',()=>{
        const wrapper=shallow(<SearchBar handleFilterTextChange={jest.fn()}/>);
        wrapper.find('.TextArea').simulate('change',{target:{value:'ball'}});
        expect(wrapper.instance().props.handleFilterTextChange).toHaveBeenCalled();
    })

    it('should trigger handleFilterTextChange on changing the textbox',()=>{
        const wrapper=shallow(<SearchBar handleInStockChange={jest.fn()}/>);
        wrapper.find('.CheckBox').simulate('change',{target:'checked'});
        expect(wrapper.instance().props.handleInStockChange).toHaveBeenCalled();
    })
    
});