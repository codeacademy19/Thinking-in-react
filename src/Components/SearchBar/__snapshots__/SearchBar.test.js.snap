// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render a Search box and a check box 1`] = `
<form>
  <input
    className="TextArea"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
  />
  <p>
    <input
      className="CheckBox"
      onChange={[Function]}
      type="checkbox"
    />
     
    Only show products in stock
  </p>
</form>
`;
